streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBStore.java

```diff
@@ -326,6 +326,7 @@ public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
 
     @Override
     public synchronized byte[] get(final Bytes key) {
+        Objects.requireNonNull(key, "key cannot be null");
```

Comment:
I think we should add this to `MeteredKeyValueStore` instead -- this way, we get the same guard for the in-memory key-value store (or other custom stores).

We should also add this check to other methods that accept a key, like `put`, `putIfAbsent` etc...

Furthermore, we should also add it for other store types, ie, `MeteredTimestampedKeyValueStore`, `MeteredWindowedStore`, `MeteredTimestampedWindowStore` and `MeteredSessionStore`.

Last, but not least, we should add corresponding unit tests.

Reply:
On it.

Reply:
This check can be remove, as we do the check in the "metered" store that will wrap this one.